### ISSUE TITLE
Library no longer works with action extension

### ISSUE THREAD

**Issue by phildow** [andreamazz/AMPopTip#73]
The inclusion of code that checks the application state (line 415 AMPopTip.m in latest build):

```
UIApplicationState state = [[UIApplication sharedApplication] applicationState];
```

assumes that the library is being used in an application when it might be being used in an action extension. Action extensions don't have access to the `UIApplication` singleton and won't compile when code includes those references.

Unfortunately this means the addition of this code precludes the use of AMPopTip in action extensions, when it worked fine in action extensions before.

One possible solution is to include a preprocessor directive that checks a build flag and conditionally compiles code that is available only to applications. This is the approach that SVProgressHUD adopts.

**Comment by andreamazz** [comment]
You are right, I was thinking 'I need to add the extension check' while writing that, then it slipped my mind 😬
I'll fix it as soon as possible

**Comment by phildow** [comment]
Thanks Andrea! Right now we're just having cocoapods keep us on build 1.0.1, so it's not too bad. =)

**Comment by andreamazz** [comment]
Fixed in `1.0.3`. You now need to specify the user macro `AM_POPTIP_EXTENSION` in your extension's target, and the forbidden API won't be called.
